ros2/test.py.tpl

```diff
@@ -18,9 +18,12 @@ sys.argv = sys.argv[:1] + [
 ] + sys.argv[1:]
 
 test_outputs_dir = os.environ.get('TEST_UNDECLARED_OUTPUTS_DIR')
+if test_outputs_dir is None:
+    test_outputs_dir = os.environ.get('TEST_TMPDIR')
 if test_outputs_dir:
-    os.environ['ROS_HOME'] = test_outputs_dir
-    os.environ['ROS_LOG_DIR'] = test_outputs_dir
+    if 'ROS_HOME' not in os.environ and 'ROS_LOG_DIR' not in os.environ:
+        os.environ['ROS_HOME'] = test_outputs_dir
+        os.environ['ROS_LOG_DIR'] = test_outputs_dir
```

Comment:
Why not treat them separately as I suggested?

Reply:
tbh I think the best way would be to set only ROS_HOME when neither ROS_HOME nor ROS_LOG_DIR is set. But as I said, this isn't the current behavior of the pytest wrapper template, and it seems like a good idea to be consistent with that